proxy/context.go

```diff
@@ -233,10 +236,66 @@ func (c *context) clone() *context {
 	return &cc
 }
 
+func (c *context) wasExecuted() bool {
+	return c.executionCounter != 0
+}
+
 func (c *context) setMetricsPrefix(prefix string) {
 	c.metrics.prefix = prefix + ".custom."
 }
 
+func (c *context) Split() (filters.FilterContext, error) {
+	originalRequest := c.Request()
+	if c.proxy.experimentalUpgrade && isUpgradeRequest(originalRequest) {
+		return nil, errors.New("context: cannot split the context that contains an upgrade request")
+	}
+	cc := c.clone()
+	cc.stateBag = map[string]interface{}{}
+	cc.responseWriter = noopFlushedResponseWriter{}
+	cc.metrics = &filterMetrics{
+		prefix: cc.metrics.prefix,
+		impl:   cc.proxy.metrics,
+	}
+	u := new(url.URL)
+	*u = *originalRequest.URL
+	u.Host = originalRequest.Host
+	cr, body, err := cloneRequest(u, originalRequest)
+	if err != nil {
+		c.proxy.log.Errorf("context: failed to clone request: %v", err)
+		return nil, err
+	}
+	serverSpan := opentracing.SpanFromContext(originalRequest.Context())
+	cr = cr.WithContext(opentracing.ContextWithSpan(cr.Context(), serverSpan))
```

Comment:
What if we would set some meta information that would show up in the new proxy Span for the shadow traffic, for example a `shadow: true` tag ?

Not sure if here or better at https://github.com/zalando/skipper/pull/1443/files#diff-969f319a10987eafd414f7c4d57e91e6R286

Reply:
I tested that it's better to have it just before finish the proxy span, such that we have the tag only in the proxy and not in the ingress span.